### PR TITLE
Txn processor tutorial and install guide for Python SDK

### PR DESCRIPTION
**Pull request by ojalatodd**
Creation of transaction processor tutorial that uses the new template system.
Produces a Python tutorial and install guide for the Python SDK, and enables
creation of tutorials for future supported languages.

### CODE REVIEW

**Comment by ojalatodd** [comment]
This PR is a combined PR that replaces #403 and #422

**Comment by ojalatodd** [comment]
IMPORTANT NOTE to all reviewers. This PR combines two other PR's that you already reviewed, so this PR shouldn't need a lot of changing (see note above). This combines #403 and #422.

**Comment by peterschwarz** [comment]
Please close #403 if it is superseded by this PR

File: docs/source/_templates/sdk_TP_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+There are two top-level components of a transaction processor: a processor
+class and a handler class. The SDK provides a general-purpose processor class.
+The handler class is application-dependent and contains the business logic for
+a particular family of transactions. Multiple handlers can be connected to an
+instance of the processor class.
+
+Handlers get called in two ways:
+
+# An ``apply`` method
+# Various "metadata" methods
```
Comment (aludvik) [inline docs/source/_templates/sdk_TP_tutorial.rst]: I think you meant to number these

Comment (ojalatodd) [inline docs/source/_templates/sdk_TP_tutorial.rst]: Fixed

**Pull request merged**